### PR TITLE
PAY-544 Card payment reconciliation impl

### PR DESCRIPTION
**Pull request by tarun-palisetty**

### CODE REVIEW

File: model/src/main/java/uk/gov/hmcts/payment/api/model/UserAwareDelegatingCardPaymentService.java [hunk 1]
```diff
@@ -91,6 +97,32 @@ public void refund(String paymentReference, int amount, int refundAmountAvailabi
...
 
+    @Override
+    @Transactional
+    public List<PaymentFeeLink> search(Date startDate, Date endDate) {
+        List<PaymentFeeLink> paymentFeeLinks = paymentFeeLinkRepository.findAll(findByDatesBetween(startDate, endDate));
+
+        // For each payment get the gov pay status.
+        paymentFeeLinks.stream().forEach(p -> {
+            Payment payment = p.getPayments().get(0);
+            GovPayPayment govPayPayment = delegate.retrieve(payment.getGovPayId());
```
Comment (jaleen) [inline model/src/main/java/uk/gov/hmcts/payment/api/model/UserAwareDelegatingCardPaymentService.java]: Please remove the gov pay check

File: model/src/main/java/uk/gov/hmcts/payment/api/model/govpay/GovPayCardPaymentService.java [hunk 2]
```diff
@@ -59,6 +61,11 @@ public void refund(@NonNull String id, int amount, int refundAmountAvailable) {
         govPayClient.refundPayment(keyForCurrentService(), hrefFor(payment.getLinks().getRefunds()), new RefundPaymentRequest(amount, refundAmountAvailable));
     }
 
+    @Override
+    public List<GovPayPayment> search(Date startDate, Date endDate) {
+        return null;
```
Comment (jaleen) [inline model/src/main/java/uk/gov/hmcts/payment/api/model/govpay/GovPayCardPaymentService.java]: How about throwing MethodNot Supported ?

Comment (tarun-palisetty) [reply model/src/main/java/uk/gov/hmcts/payment/api/model/govpay/GovPayCardPaymentService.java]: This method has nothing to do in search, so it is left unimplemented.

File: model/src/main/java/uk/gov/hmcts/payment/api/model/UserAwareDelegatingCardPaymentService.java (hunk 1 above)
Comment (tarun-palisetty) [reply model/src/main/java/uk/gov/hmcts/payment/api/model/UserAwareDelegatingCardPaymentService.java]: Comment out the code.

**Pull request merged**